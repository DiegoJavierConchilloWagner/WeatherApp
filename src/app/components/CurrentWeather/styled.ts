import styled from 'styled-components';

export const StyledCurrentWeather = styled.div`
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	flex-wrap: wrap;
	height: fit-content;
	margin: 0 10vw;
	color: ${({ theme }) => theme.colors.typography.primary};
	font-family: ${({ theme }) => theme.fonts.primary};
	gap: 2rem;

	@media (max-width: 768px) {
		margin: 3rem 10vw;
		gap: 2rem;
	}
`;

export const InfoContainer = styled.div`
	position: relative;
	display: flex;
	flex-direction: row;
	align-items: center;
	gap: 3rem;

	@media (max-width: 768px) {
		gap: 1.5rem;
		flex-direction: row;
		align-items: center;
	}

	div {
		h1 {
			font-weight: 700;
			font-size: 2.5rem;

			@media (max-width: 768px) {
				font-size: 2rem;
			}
		}
		p {
			margin-top: 0px;
			font-weight: 500;

			@media (max-width: 768px) {
				font-size: 0.8rem;
			}
		}
	}
	h3 {
		font-size: 2rem;
		font-weight: 600;

		@media (max-width: 768px) {
			font-size: 1.5rem;
		}
	}
`;

export const WeatherContainer = styled.div`
	display: flex;
	width: 20rem;
	flex-direction: column;
	align-items: center;
	gap: 1rem;

	@media (max-width: 768px) {
		justify-content: center;
		gap: 0.2rem;
	}

	p {
		align-self: center;
		font-weight: 700;
		border: 1px solid white;
		border-top-right-radius: 40px;
		border-bottom-left-radius: 40px;
		width: 7rem;
		text-align: center;
	}

	h5 {
		font-size: 1.2rem;
		font-weight: 400;
	}
`;

export const TemperatureContainer = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;

	@media (max-width: 768px) {
		flex-direction: column;
		justify-content: center;
		text-align: center;
	}

	div {
		margin-right: 0rem;

		@media (max-width: 768px) {
			margin-right: 0%;
			align-items: center;
		}

		h4 {
			span {
				font-weight: 400;
				font-size: 1.2rem;

				@media (max-width: 768px) {
					font-size: 1rem;
				}
			}
			font-weight: 600;
			font-size: 1.5rem;

			@media (max-width: 768px) {
				font-size: 1.2rem;
			}
		}

		h5 {
			font-weight: 400;
			font-size: 1.2rem;
			text-align: center;

			@media (max-width: 768px) {
				font-size: 1.5rem;
			}

			span {
				font-weight: 400;
				font-size: 1.2rem;

				@media (max-width: 768px) {
					font-size: 1rem;
				}
			}
		}
	}
`;
