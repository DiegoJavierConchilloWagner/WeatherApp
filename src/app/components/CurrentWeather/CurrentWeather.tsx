import { FC } from 'react';

import dayjs from 'dayjs';

import { useSelector } from 'react-redux';

import { kelvinToCelcius } from '@common/index';
import { RootState } from '@redux/store/store';

import { GeoError } from '@components/ui/Errors';
import { WeatherIcon } from '../WeatherIcon/WeatherIcon';
import { InfoBLock, InfoContainer, StyledCurrentWeather, TemperatureContainer, WeatherContainer } from './styled';

interface Props {
	isLoading: boolean;
	isError: boolean;
}

export const CurrentWeather: FC<Props> = ({ isLoading, isError }) => {
	const { current } = useSelector((state: RootState) => state.WeatherInterface);
	const { selectedPlace } = useSelector((state: RootState) => state.placeInterface);
	const { isCurrentWeatherLoading } = useSelector((state: RootState) => state.uiInterface);

	const { dt, feelsLike, temp, main, id, icon } = current;
	const date = dayjs(new Date(dt * 1000));

	const selectedPlaceName =
		selectedPlace.name.length > 0 ? selectedPlace.name.split(',')[0] : selectedPlace.name.split(',')[1];
	const currentDate = dt ? date.format('MMM, dddd YYYY') : 'Loading..';
	const currentHour = dt ? date.format('HH:mm') : '';
	const tempFeelsLike = feelsLike ? kelvinToCelcius(feelsLike) : '..';
	const currentTemp = temp ? kelvinToCelcius(temp) : '..';

	if (isError) return <GeoError />;

	return (
		<StyledCurrentWeather>
			<InfoContainer>
				<InfoBLock>
					<h1>{selectedPlaceName || 'Curent Location'}</h1>
					<p>{currentDate}</p>
				</InfoBLock>
				<h3>{currentHour}</h3>
			</InfoContainer>
			<WeatherContainer>
				<p>Today</p>
				<TemperatureContainer>
					<div>
						<WeatherIcon id={id} icon={icon} size={100} isLoading={isLoading} />
						{!isCurrentWeatherLoading && <h5>{main}</h5>}
					</div>
					<div>
						<h4>
							<span>Feels-Like:</span> {isCurrentWeatherLoading ? '..' : tempFeelsLike}°
						</h4>
						<h5>
							<span>Temperature:</span> {isCurrentWeatherLoading ? '..' : currentTemp}°
						</h5>
					</div>
				</TemperatureContainer>
			</WeatherContainer>
		</StyledCurrentWeather>
	);
};
