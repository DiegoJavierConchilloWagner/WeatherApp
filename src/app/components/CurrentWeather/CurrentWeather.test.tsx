import renderWithThemeAndProvider from '@common/utils/renderWhitThemeAndProvider';
import { CurrentWeather } from './CurrentWeather';

describe('Example CurrentWeather', () => {
	it('should render correcty', () => {
		const wrapper = renderWithThemeAndProvider(<CurrentWeather isLoading={false} isError={false} />);

		expect(wrapper).toMatchSnapshot();
	});
});
