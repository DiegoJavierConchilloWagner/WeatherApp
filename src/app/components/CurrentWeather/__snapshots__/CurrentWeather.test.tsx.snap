// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example CurrentWeather should render correcty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  margin: 0 10vw;
  color: #fff;
  font-family: Poppins;
  gap: 2rem;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  gap: 3rem;
}

.c1 h3 {
  font-size: 2rem;
  font-weight: 600;
}

.c2 h1 {
  font-weight: 700;
  font-size: 2.5rem;
}

.c2 p {
  margin-top: 0px;
  font-weight: 500;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 20rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
}

.c3 p {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  font-weight: 700;
  border: 1px solid white;
  border-top-right-radius: 40px;
  border-bottom-left-radius: 40px;
  width: 7rem;
  text-align: center;
}

.c3 h5 {
  font-size: 1.2rem;
  font-weight: 400;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 div {
  margin-right: 0rem;
}

.c4 div h4 {
  font-weight: 600;
  font-size: 1.5rem;
}

.c4 div h4 span {
  font-weight: 400;
  font-size: 1.2rem;
}

.c4 div h5 {
  font-weight: 400;
  font-size: 1.2rem;
  text-align: center;
}

.c4 div h5 span {
  font-weight: 400;
  font-size: 1.2rem;
}

@media (max-width:768px) {
  .c0 {
    margin: 3rem 10vw;
    gap: 2rem;
  }
}

@media (max-width:768px) {
  .c1 {
    gap: 1.5rem;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:768px) {
  .c1 h3 {
    font-size: 1.5rem;
  }
}

@media (max-width:768px) {
  .c2 h1 {
    font-size: 2rem;
  }
}

@media (max-width:768px) {
  .c2 p {
    font-size: 0.8rem;
  }
}

@media (max-width:768px) {
  .c3 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    gap: 0.2rem;
  }
}

@media (max-width:768px) {
  .c4 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c4 div {
    margin-right: 0%;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:768px) {
  .c4 div h4 span {
    font-size: 1rem;
  }
}

@media (max-width:768px) {
  .c4 div h4 {
    font-size: 1.2rem;
  }
}

@media (max-width:768px) {
  .c4 div h5 {
    font-size: 1.5rem;
  }
}

@media (max-width:768px) {
  .c4 div h5 span {
    font-size: 1rem;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <h1>
              Curent Location
            </h1>
            <p>
              Loading..
            </p>
          </div>
          <h3 />
        </div>
        <div
          class="c3"
        >
          <p>
            Today
          </p>
          <div
            class="c4"
          >
            <div>
              <img
                alt="weather"
                height="100"
                src="SvgrURL"
                width="100"
              />
              <h5 />
            </div>
            <div>
              <h4>
                <span>
                  Feels-Like:
                </span>
                 
                ..
                째
              </h4>
              <h5>
                <span>
                  Temperature:
                </span>
                 
                ..
                째
              </h5>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  margin: 0 10vw;
  color: #fff;
  font-family: Poppins;
  gap: 2rem;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  gap: 3rem;
}

.c1 h3 {
  font-size: 2rem;
  font-weight: 600;
}

.c2 h1 {
  font-weight: 700;
  font-size: 2.5rem;
}

.c2 p {
  margin-top: 0px;
  font-weight: 500;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 20rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
}

.c3 p {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  font-weight: 700;
  border: 1px solid white;
  border-top-right-radius: 40px;
  border-bottom-left-radius: 40px;
  width: 7rem;
  text-align: center;
}

.c3 h5 {
  font-size: 1.2rem;
  font-weight: 400;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 div {
  margin-right: 0rem;
}

.c4 div h4 {
  font-weight: 600;
  font-size: 1.5rem;
}

.c4 div h4 span {
  font-weight: 400;
  font-size: 1.2rem;
}

.c4 div h5 {
  font-weight: 400;
  font-size: 1.2rem;
  text-align: center;
}

.c4 div h5 span {
  font-weight: 400;
  font-size: 1.2rem;
}

@media (max-width:768px) {
  .c0 {
    margin: 3rem 10vw;
    gap: 2rem;
  }
}

@media (max-width:768px) {
  .c1 {
    gap: 1.5rem;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:768px) {
  .c1 h3 {
    font-size: 1.5rem;
  }
}

@media (max-width:768px) {
  .c2 h1 {
    font-size: 2rem;
  }
}

@media (max-width:768px) {
  .c2 p {
    font-size: 0.8rem;
  }
}

@media (max-width:768px) {
  .c3 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    gap: 0.2rem;
  }
}

@media (max-width:768px) {
  .c4 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c4 div {
    margin-right: 0%;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:768px) {
  .c4 div h4 span {
    font-size: 1rem;
  }
}

@media (max-width:768px) {
  .c4 div h4 {
    font-size: 1.2rem;
  }
}

@media (max-width:768px) {
  .c4 div h5 {
    font-size: 1.5rem;
  }
}

@media (max-width:768px) {
  .c4 div h5 span {
    font-size: 1rem;
  }
}

<div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <h1>
            Curent Location
          </h1>
          <p>
            Loading..
          </p>
        </div>
        <h3 />
      </div>
      <div
        class="c3"
      >
        <p>
          Today
        </p>
        <div
          class="c4"
        >
          <div>
            <img
              alt="weather"
              height="100"
              src="SvgrURL"
              width="100"
            />
            <h5 />
          </div>
          <div>
            <h4>
              <span>
                Feels-Like:
              </span>
               
              ..
              째
            </h4>
            <h5>
              <span>
                Temperature:
              </span>
               
              ..
              째
            </h5>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
